core/raft/queue/opqueue.go

```diff
@@ -112,6 +128,10 @@ const (
 )
 
 func (q *OpQueue) loop() error {
+	// Ensure we close the queue channels when we've been killed.
+	defer close(q.in)
```

Comment:
We should generally avoid closing input channels on the receiving end. How do we guarantee that nothing tries to write to `q.in` when loop returns?

Reply:
Yeah legit.